src/primitives/cone.jl

```diff
@@ -0,0 +1,30 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+  Cone(vertex, center, radius, nsides, closed)
+
+Cone with `vertex` and base with `center` and `radius` made of `nsides` 
+segments. If `closed`, the base is included.
+"""
+struct Cone{T,I<:Integer} <: Primitive{3,T}
+  vertex::Point{3,T}
+  center::Point{3,T}
+  radius::T
+  nsides::I
+  closed::Bool
+end
+
+function Cone(vertex::Tuple, center::Tuple, radius, nsides, closed)
```

Comment:
Take a look at Sphere for an example of outer constructors. We first need to make sure that the outer constructor with geometry types has the correct result and then we can wrap it in a second outer constructor with tuples.